### PR TITLE
Content header title length

### PR DESCRIPTION
**Pull request by davidcmoulton**
Implements https://github.com/elifesciences/pattern-library/pull/919/files

### CODE REVIEW

File: src/ViewModel/ContentHeader.php [hunk 1]
```diff
@@ -78,6 +85,31 @@ public function __construct(
         $this->audioPlayer = $audioPlayer;
     }
 
+    public function designateTitleLength() : string
```
Comment (thewilkybarkid) [inline src/ViewModel/ContentHeader.php]: Needs to be `private`. Cleaner to take the title as input too.

File: src/ViewModel/ContentHeader.php [hunk 2]
```diff
@@ -78,6 +85,31 @@ public function __construct(
...
+    {
+        $charCount = strlen(strip_tags($this->title));
```
Comment (thewilkybarkid) [inline src/ViewModel/ContentHeader.php]: `strlen` actually counts the number of bytes, we need to use `mb_strlen`.

File: src/ViewModel/ContentHeader.php [hunk 3]
```diff
@@ -78,6 +85,31 @@ public function __construct(
...
+            return self::MEDIUM;
+        }
+        if ($charCount <= 118) {
+            return self::LONG;
+        }
+        if ($charCount <= 155) {
+            return self::X_LONG;
+        }
+
+        return self::XX_LONG;
```
[the remark this answers is not in the document]
Comment (davidcmoulton) [reply src/ViewModel/ContentHeader.php]: 🤦‍♂️ obvious now you mention it 👍

File: src/ViewModel/ContentHeader.php [hunk 4]
```diff
@@ -26,13 +26,16 @@
...
+    const LENGTH_LIMITS = [
+        19 => 'xx-short',
+        38 => 'x-short',
+        46 => 'short',
+        57 => 'medium',
+        118 => 'long',
+        155 => 'x-long',
+    ];
+
+    const MAX_LENGTH = 'xx-long';
```
Comment (davidcmoulton) [inline src/ViewModel/ContentHeader.php]: Separated `MAX_LENGTH` from the `LENGTH_LIMITS` array as it's the only one not designated by a maximum character count.

File: tests/src/ViewModel/ContentHeaderTest.php [hunk 5]
```diff
@@ -217,6 +233,133 @@ public function institutions_must_be_institutions()
...
+     */
+    public function a_title_longer_than_155_characters_is_xx_long(string $title)
+    {
+        $this->assertGreaterThan(155, strlen($title));
+
+        $contentHeader = self::buildFixtureForCollection($title);
+        $this->assertSame('xx-long', $contentHeader['titleLength']);
+    }
+
+    public function xx_long_title_provider() : array
```
Comment (davidcmoulton) [inline tests/src/ViewModel/ContentHeaderTest.php]: @thewilkybarkid Is this how you envisaged the use of a `dataProvider` here?

File: src/ViewModel/ContentHeader.php [hunk 6]
```diff
@@ -78,6 +88,19 @@ public function __construct(
         $this->audioPlayer = $audioPlayer;
     }
 
+    private static function designateTitleLength($title) : string
```
Comment (davidcmoulton) [inline src/ViewModel/ContentHeader.php]: @thewilkybarkid `CarouselItem` needs to use this, too. Do you think `designateTitleLength()` should be extracted, along with its associated `const`s `LENGTH_LIMITS` and `MAX_LENGTH` into a `trait`?

**Pull request merged**